factcheck/core/Retriever/serper_retriever.py

```diff
@@ -85,13 +86,17 @@ def _retrieve_evidence_4_all_claim(self, query_list: list[str], top_k: int = 5,
                     }
             else:
                 results = result.get("organic", [])[:top_k]  # Choose top 5 result
-                merge_evidence_text = [f"Text: {_result['snippet']} \n Source: {_result['link']}" for _result in results]
-                merge_evidence_text = [re.sub(r"\n+", "\n", evidence) for evidence in merge_evidence_text]
+                merge_evidence_text = [
+                    f"Text: {_result['snippet']} \n Source: {_result['link']} \n Date: {_result.get('date', 'Unknown')}"
```

Comment:
Include the date of the result in the prompt. Thanks to this, OFV can verify claims such as "Did X happen before Y?"

Reply:
Hi! The inclusion of the date as a separate factor might be helpful, but I suspect it is not essential. In your example, "X happens before Y," we would expect the generated queries to be like ["When did X happen?", "When did Y happen?"]. Given such queries, retrieved evidence is expected to be sufficient to solve this problem. 

Prompt for query generation: 
https://github.com/Libr-AI/OpenFactVerification/blob/f620f6e5879ee1219661afc749c9fd8f658b194b/factcheck/utils/prompt/chatgpt_prompt.py#L43C1-L62

Reply:
Hey, sorry for late reply, I was on vacation.

Unfortunately, not all articles come with clear dates in them. But Serp can return their creation date anyway.

Reply:
Maybe this could be available behind a flag in `FactCheck`'s initialisation?